app/team.go

```diff
@@ -223,8 +227,17 @@ func (a *App) AddUserToTeamByHash(userId string, hash string, data string) (*mod
 		return nil, model.NewAppError("JoinUserToTeamByHash", "api.user.create_user.signup_link_invalid.app_error", nil, "", http.StatusBadRequest)
```

Comment:
Why are we still doing hash verification? We should just use the token if we are going to add that.

Reply:
We still need to send data like the email (shared with the webapp) so I wanted to maintain the changes small. Besides the quantity of extra data that can be included in the token is only 128 bytes, so we can't include all the current data in the token.

Reply:
I would rather remove the extra complexity. You don't need to have verified all the data that we currently send down. Only the stuff we actually trust needs to be stored in the token. (so userId and teamId, the rest is just for show)

Reply:
Yes, you are right, probably we don't need that complexity.